website/docs/r/dse_registry_configuration.html.markdown

```diff
@@ -0,0 +1,152 @@
+sidebar_current: "docs-vcd-resource-dse-registry-configuration"
+description: |-
+  Provides a resource to manage Data Solution Extension (DSE) registry configuration.
+---
+
+# vcd\_dse\_registry\_configuration
+
+Supported in provider *v3.13+* with Data Solution Extension.
+
+Provides a resource to manage Data Solution Extension (DSE) registry configuration.
+
+~> Only `System Administrator` can create this resource.
+
+## About Data Solution structure
+
+There are 2 types of Data Solution configurations:
+* Helm based
+* Container based
+
+All of Data Solutions provide default versions and repositories that can be used by setting
+`use_default_value` flag that will set these defaults in configuration.
+
+One can also set custom repositories, package names and their versions. For Helm based configuration
+one should specify `chart_repository`, `version` and `package_name`. For container based
+configuration, `package_repository` and `version` should be set.
+
+Helm based configurations apply to:
+
+* `MongoDB Community`
+* `Confluent Platform`
+* `MongoDB`
+
+
+Container based configurations apply to:
+
+* `VMware SQL with Postgres`
+* `VMware RabbitMQ`
+* `VMware SQL with MySQL`
+* `VCD Data Solutions`
+
+Additionally, `VCD Data Solutions` provide [Container Registry Configuration ](#container-registry).
+
+## Example Usage (Configure chart repository)
+
+```hcl
+resource "vcd_dse_registry_configuration" "mongodb-community" {
+  name             = "MongoDB Community"
+  chart_repository = "https://mongodb.github.io/helm-charts"
+  version          = "0.9.0"
+  package_name     = "community-operator"
+
+  depends_on = [vcd_solution_add_on_instance_publish.public]
+}
+```
+
+## Example Usage (Configure package repository)
+
+```hcl
+resource "vcd_dse_registry_configuration" "mysql" {
+  name               = "VMware SQL with MySQL"
+  package_repository = "registry.tanzu.vmware.com/packages-for-vmware-tanzu-data-services/tds-packages:1.13.0"
+  version            = "1.10.1"
+
+  depends_on = [vcd_solution_add_on_instance_publish.public]
+}
+```
+
+## Example Usage (Configure container registries)
+
+```hcl
+resource "vcd_dse_registry_configuration" "dso" {
+  name = "VCD Data Solutions"
+
+  # Using default versions for packages
+  use_default_value = true
+
+  container_registry {
+    host        = "first-host.sample"
+    description = "host2"
+    username    = "user1"
+    password    = "pass1"
+  }
+
+  container_registry {
+    host        = "another-host.sample"
+    description = "Test Host that does not work"
+    username    = "user2"
+    password    = "pass2"
+  }
+
+  depends_on = [vcd_solution_add_on_instance_publish.public]
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Required) The name of Data Solution as it appears in repository configuration
+* `use_default_value` - (Optional) Defines if repository settings should be inherited from Data
```

Comment:
Reading with fresh eyes this is obscure. I'm thinking what if it were named 'use_default_version' instead?

Reply:
It is not only the version, it is also repository address details (all the fields you can see as greyed out in these screenshots)

<img width="911" alt="image" src="https://github.com/vmware/terraform-provider-vcd/assets/15804230/7a1cfc3a-a166-4396-874d-a30c92f8faaa">
<img width="829" alt="image" src="https://github.com/vmware/terraform-provider-vcd/assets/15804230/69d64bd3-013e-4524-9fee-75507b7693da">